internal/stats/stats.go

```diff
@@ -0,0 +1,43 @@
+package stats
+
+import (
+	"log"
+	"sort"
+	"sync"
+)
+
+var (
+	mu     sync.Mutex
+	counts = make(map[string]int64)
```

Comment:
@jules this should not be a global it should be in a module or stored in Server accessible via CD. refactor out all stats related globals